### ISSUE TITLE
Feature - Security Scan 

### ISSUE THREAD

**Issue by naveensrinivasan** [ossf/scorecard#966]
**Is your feature request related to a problem? Please describe.**
Should Scorecard run a security scan on repositories? With something like https://github.com/coinbase/salus

**Comment by chrismcgehee** [comment]
Here's a question I would ask: Do the raw results of an automated security scan provide a meaningful insight as to the security of a project? I would argue it does not due to the high number of false positives in SAST tools.

**Comment by naveensrinivasan** [comment]
IMO gosec provides good security defaults. 

The whole idea is to make OSS aware of security implications that most projects aren’t providing. The goal is to encourage them to address these. 

Most of these linters have an option nolint, which meant the owner of the repository took their time to address those false positives even if it is.

**Comment by chrismcgehee** [comment]
I've thought about this further, and I think this is worth doing.

**Comment by laurentsimon** [comment]
there are various shades of grey we may consider here (sorry was OOO yesterday):
1. Check for specific SAST tools. As simple as looking for bot names run on PR and/or parsing config files (what David said)
2. Classify *what* guarantees a SAST tool provides. Example: [go-kart](https://github.com/praetorian-inc/gokart) tries to detect command Injection, path traversal. I don't think `gosec` does. But gosec does a good job at checking for bad crypto API (e.g., md5).  This is value to understanding what we get from a particular tool. All this is time-consuming and the tools may change over time. 
3. Surfacing SAST results in scorecard. The logic for this cannot live in the scorecard codebase because it requires running 3P tools, would require sandboxing on user's machines, etc. Long-term however, we can build this as a service. This has the advantages of:
3.1 Letting repo users read the results without installation.
3.2 Letting everyone read the results, which is currently not possible in GitHub scans (they are private to maintainers only)
3.3 Result can be surfaced in scorecard via an REST API to the service - similar to what we do for OSV.

One big blocker is false positive, like already mentioned. At this point we need to think about how we handle annotation: language-specific vs language-agnostic? Then, to remove FPs, we could proceed in stages:
1. Trust the repo owner to add annotation themselves
2. Have an army of security consultant who can vet the annotation. Scorecard can check *who* accepted the change.

With a service, we could, long-long-term:
1. Automatically file issues on the repo
2. Wait X days for a fix or an annotation
3. After X days have passed, file an OSV bug which can easily be surfaced in the `Vulnerabilities` check.

**Comment by chrismcgehee** [comment]
I like what you've put down here. I just wanted to note the things you put under long-long term sounds like the things Project Omega from Michael Scovetta aims to do.

**Comment by laurentsimon** [comment]
do you have a link to the project?

**Comment by chrismcgehee** [comment]
Yes, https://docs.google.com/document/d/1u7Ps18dzu9M-HF7ZHTK6VB5jLaVJvnw6uq3o7qw5yGE/edit?usp=sharing

**Comment by laurentsimon** [comment]
also related https://ostif.org/google-is-partnering-with-open-source-technology-improvement-fund-inc-to-sponsor-security-reviews-of-critical-open-source-software/

**Comment by laurentsimon** [comment]
cc @scovetta this thread looks awfully similar to alpha/omega project, as @chrismcgehee pointed out. Let's try to ingest their data in scorecard when it's ready; or even work with them.
